hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -62,6 +62,11 @@ object DataSourceReadOptions {
       "(or) Read Optimized mode (obtain latest view, based on base files) (or) Snapshot mode " +
       "(obtain latest view, by merging base and (if any) log files)")
 
+  val QUERY_USE_DATABASE: ConfigProperty[Boolean] = ConfigProperty
+    .key("hoodie.query.use.database")
```

Comment:
sorry, I am a little confused about the config and the use case of this config.

Reply:
@leesf  This configuration is reflected in the test case. The main consideration is that if different databases have the same table name, such as db1.table1 and db2.table1, and if the two tables are queried in the same session at the same time, I only want to set the incremental query parameters of db1.table1：
```sql
set hoodie.table1.datasource.query.type=incremental;
set hoodie.table1.datasource.read.begin.instanttime=20221130163703640;
```
In this way, although I only want to query db1.table1 incrementally, I will also perform incremental queries when querying db2.table1. This is not the effect I expected, so I have this parameter：
```sql
set hoodie.query.use.database = true;
set hoodie.db1.table1.test_hudi_incremental.datasource.query.type=incremental;
set hoodie.db1.table1.test_hudi_incremental.datasource.read.begin.instanttime=20221130163703640;
```
In this way, we can only perform incremental queries on db1.table1. This configuration is false by default, which is consistent with the Hive incremental query parameters

Reply:
The PR of Hive incremental query：[https://github.com/apache/hudi/pull/4083](https://github.com/apache/hudi/pull/4083)

Reply:
If it only affect incremental query, maybe `hoodie.query.incremental.database` is a better name? or it is also affect other types of query? then we need to add more test cases.

Reply:
It also affects other types of queries. I can add test cases of other query types.

Reply:
@leesf Hello, I have added test cases of other query types